contrib/registry/consul/client.go

```diff
@@ -33,9 +41,14 @@ type Client struct {
 	serviceChecks api.AgentServiceChecks
 }
 
+type ClientConfig struct {
+	DCMode Datacenter
+}
+
 // NewClient creates consul client
-func NewClient(cli *api.Client) *Client {
+func NewClient(cli *api.Client, cfg *ClientConfig) *Client {
```

Comment:
This method should be marked as obsolete and need not be provided for external use